### PR TITLE
Dangermattic Setup

### PR DESCRIPTION
**Pull request by iangmaia**
Setting up the use of [Dangermattic](https://github.com/Automattic/dangermattic) checks in the project, replacing the [Peril checks](https://github.com/Automattic/peril-settings/blob/master/peril-settings.json).

These are the checks currently implemented for WPAndroid:
- Labels (equivalent to [label.ts](https://github.com/Automattic/peril-settings/blob/master/org/pr/label.ts))
- PR Size (equivalent to [android-diff-size.ts](https://github.com/Automattic/peril-settings/blob/master/org/pr/android-diff-size.ts))
- Milestone (equivalent to [milestone.ts](https://github.com/Automattic/peril-settings/blob/master/org/pr/milestone.ts))
- View Changes Need Screenshots (new check)
- Manifest / Lockfile (new check)
- Android unit tests missing (new check)
- Rubocop via Danger

**NOTE**: I also have an additional PR, https://github.com/wordpress-mobile/WordPress-Android/pull/19218, which improves the project's Rubocop setup -- that is the reason there's a Danger failure being reported on CI.

##  How to test

You should see a comment stating the warnings / errors present in this PR. As a test, I've also added a "do not merge" label so that Danger also issues an error, and that error should also be reported on CI.

To run Danger yourself:
1. Checkout this branch 
2. Run `bundle install`
3. Run:
```ruby
DANGER_GITHUB_API_TOKEN=<github_token> bundle exec danger pr <PR_URL>
```
You should get in the console the same errors / warnings reported in this PR.

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+return if github.pr_labels.include?('Releases')
+
+github.dismiss_out_of_range_messages
+
+manifest_pr_checker.check_gemfile_lock_updated
+
+labels_checker.check(
+  required_labels: [//],
+  required_labels_error: 'PR requires at least one label.'
+)
```
Comment (mokagio) [inline Dangerfile:27]: Did you consider adding support for "do not merge" here, even though it wasn't there before?

File: Dangerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+pr_size_checker.check_diff_size(
+  file_selector: ->(path) { !path.include?('/src/test') },
+  max_size: 300
+)
+
+android_unit_test_checker.check_missing_tests
+
+# skip check for draft PRs and for WIP features unless the PR is against the main branch or release branch
+milestone_checker.check_milestone_due_date(days_before_due: 4) if !github.pr_draft? && (!wip_feature? || (release_branch? || main_branch?))
```
Comment (mokagio) [inline Dangerfile]: What do you think of using positive logic and using the same wording as the comment in the WIP... logic?

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 4) unless github.pr_draft? || (wip_feature? && !(release_branch? || main_branch?))
```

File: Dangerfile (hunk 1 above)
Comment (iangmaia) [reply Dangerfile:27]: The plugin has a parameter for a "do not merge label", and the default is `Do Not Merge` (case insensitive) which matches several of our projects. So if you don't specify anything, it's a check you'll get -- some projects (e.g. [Woo iOS](https://github.com/woocommerce/woocommerce-ios/pull/10654/files#diff-c429cc891087188b07403da9305b15810b0b14e3ad75cef11bdf09580c4734ddR25)) use a different label, and in those cases we need to specify the parameter.

The [Peril configuration](https://github.com/Automattic/peril-settings/blob/2fb61517b92205323c96c30b90bb4307ef6b8f13/org/pr/label.ts#L13)  actually does something like that as well.

File: Dangerfile (hunk 2 above)
Comment (iangmaia) [reply Dangerfile]: That's a great suggestion -- I actually had a lot of trouble parsing the original rule in Peril, this improves a lot the readability, going together with the comment.

File: Dangerfile (hunk 1 above)
Comment (mokagio) [reply Dangerfile:27]: Gotcha.

🤔 Shame on me for forgetting about this. But it does reveal a somewhat hidden default behavior. All good in the context of a tool for internal usage.

**Review by mokagio — APPROVED** [review]

File: Dangerfile (hunk 1 above)
Comment (iangmaia) [reply Dangerfile:27]: > But it does reveal a somewhat hidden default behavior. 

Very true -- I also noticed that some of these defaults might hide too much what is going to happen. It's worth it taking a second look at them in any case.

Comment (AliSoftware) [reply Dangerfile:27]: > But it does reveal a somewhat hidden default behavior. All good in the context of a tool for internal usage.

@mokagio We really do have the same mind 😂 

https://github.tumblr.net/TumblrMobile/android2/pull/21109#discussion_r472272
https://github.tumblr.net/TumblrMobile/android2/pull/21109#discussion_r472275

Comment (iangmaia) [reply Dangerfile:27]: > @mokagio We really do have the same mind 😂

Ha, yes 😂 

By the way, I forgot to mention there's [an issue](https://github.com/orgs/Automattic/projects/775/views/1?pane=issue&itemId=42524204) about the default values that I created when @AliSoftware pointed out the same.

Comment (mokagio) [reply Dangerfile:27]: @AliSoftware @iangmaia 😄 👌

**Pull request merged**